test/cpp_extensions/mtia_extension.cpp

```diff
@@ -38,9 +38,8 @@ struct MTIAGuardImpl final : public c10::impl::DeviceGuardImplInterface {
   }
 
   void setDevice(c10::Device d) const override {
-    c10::Device current_device = getDevice();
-    if (current_device.index() != d.index()) {
-      current_device = d;
+    if (getDevice().index() != d.index()) {
+      current_device = d.index();
```

Comment:
we should rename it to curr_device_idx or something more readable, but that can be done outside of this fix. thanks!